### ISSUE TITLE
API exception while reading disassembly_text

### ISSUE THREAD

**Issue by janfrancu** [Vector35/binaryninja-api#983]
I have caught the following type of exception `-846058720 is not a valid InstructionTextTokenContext` while reading disassembly text using
```
try:
    dtext = b.disassembly_text
except Exception as e:
    raise Exception("Encountered " + str(e) + " at block " + str(b))
```
The exact number -846058720 is not important, as it changes with each run. I have encountered this yet with two binaries `14635dfcc132e611155d276670d9603bb064eced4cf429e5112272c5f360205a` , `58af924e90c5a8f95e663d63b4058a2d963bc9b88463bff699e431961eca8222`, which can be found in the VirusTotal database. Note, that I am on the stable branch.

When looking at the affected blocks in the UI, where it works just fine, I see this.

![58af924e90c5a8f95e663d63b4058a2d963bc9b88463bff699e431961eca8222](https://user-images.githubusercontent.com/34743109/37392619-ed1e34c2-276e-11e8-8f7b-abab5b7268c4.png)

![14635dfcc132e611155d276670d9603bb064eced4cf429e5112272c5f360205a](https://user-images.githubusercontent.com/34743109/37392620-ed3cbadc-276e-11e8-91e5-a7e3fec46b3f.png)

**Comment by janfrancu** [comment]
I have investigated this issue on the newest dev version and it seems to be resolved. The difference is now that the blocks do not end with the { Falls through ... , where, according to my debugging, the possible address token's context was causing the exception. In the above cases the token had the following fields
`(<InstructionTextTokenType.PossibleAddressToken: 5>, '0x544ca7', 5524647L, 4L, 140729011989288L, 49049648, 255, 49049536L)`
and
 `(<InstructionTextTokenType.PossibleAddressToken: 5>, '0x402ad0', 4205264L, 4L, 140736583801688L, 18200064, 255, 18199952L)`
corresponding to `(token_type, text, value, size, operand, context, confidence, address)`.

Version 1.1.1225-dev the disassembly output:
![58af924e90c5a8f95e663d63b4058a2d963bc9b88463bff699e431961eca8222-dev](https://user-images.githubusercontent.com/34743109/40975989-f79d947a-68cc-11e8-9b18-08dbca102939.png)
![14635dfcc132e611155d276670d9603bb064eced4cf429e5112272c5f360205a-dev](https://user-images.githubusercontent.com/34743109/40975990-f7bd95b8-68cc-11e8-922c-8b930ee9024d.png)

I will try to run my analysis through another bunch of binaries (around 500 samples) that failed this way and if everything goes well, I think the issue can be closed.

**Comment by plafosse** [comment]
Any update on the results of this?

**Comment by janfrancu** [comment]
Spent with this whole yesterday and much of today... From the samples that I have seen and it seems to be significantly remedied, at least for the `int3` cases. However I have found two cases of blocks ending with common instructions like `cmp eax, 0xc0000225` or `pop ecx`, where the PossibleAddressToken's context is still incorrect. Unfortunately I cannot share these binaries, so pictures must suffice.

cmp case
![cmp_case](https://user-images.githubusercontent.com/34743109/42088521-0669faea-7b9a-11e8-8b46-a34df994720f.png)
![cmp_case_lindis](https://user-images.githubusercontent.com/34743109/42088522-06863c6e-7b9a-11e8-8c98-e2452ca309a4.png)

pop case
![pop_case](https://user-images.githubusercontent.com/34743109/42088523-06a1b4d0-7b9a-11e8-9e7f-97646224c8dc.png)
![pop_case_lindis](https://user-images.githubusercontent.com/34743109/42088524-06bba566-7b9a-11e8-982c-3e567380a31b.png)

I have included the linear disassembly to show the subsequent bytes. Also one observation I have made is that the context changes in each run only in the last 6 digits.

As I said at the beginning the problem now does not occur in the `int3` cases, which are more prevalent (something like 99%). From my point of view, it is definitely an improvement, almost to a point, that I consider it negligible, however if it signifies some deeper internal, we should probably leave it open.
